### PR TITLE
Documentation for the new-style testing approach

### PR DESCRIPTION
**Pull request by floehopper**
I've had a first stab at this, because @pmanrubia asked about it in the context of the new version of the minimum wage flow - see #2390.

I don't think we've written this down anywhere else - please correct me if I'm wrong.

@chrisroos: I'd definitely like your feedback on this.

### CODE REVIEW

File: doc/new-style-testing.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+* Faster test suite
+
+## Old style
+
+* Previously most of the test coverage for a flow was provided by a single integration test often with [lots of nested contexts](testing.md#deprecated-a-test-using-nested-contexts).
+* Flows that have a calculator also tend to have a unit test for that calculator.
+* When we started doing significant refactoring of the app, we weren't happy that we had sufficient test coverage to support this work.
+* Notably none of the tests actually *rendered* the question or outcome pages.
+* So at this point we introduced the [regression tests](regression-tests.md) as a relatively quick way to improve the test coverage.
+* However, the intention has always been that these regression tests are a a only temporary measure.
```
Comment (chrisroos) [inline doc/new-style-testing.md]: Typo: "...are a a only temporary..."

**Comment by chrisroos** [comment]
I'd be inclined to swap the order of the "New style" and "Old style" sections of the document. I found it confusing that the introduction talks about a new style of testing but then tells me about the old style.

Aside from that, this looks great to me!

File: doc/new-style-testing.md (hunk 1 above)
Comment (floehopper) [inline doc/new-style-testing.md]: Fix-up commit: 7b0be62f41e5f505ba70b488a6cea79d3e1557bc

**Comment by floehopper** [comment]
I've swapped the old-style and new-style sections as suggested by @chrisroos. Fix-up commit: dbd7f898875cfd47ebb39b9dbaef5118b92ec129

**Comment by floehopper** [comment]
Rebasing to include fixup commits, then rebase against master and force-push in preparation for merging.

**Pull request merged**